### PR TITLE
Gsoc2023, stand-alone interactive program that demonstrates 2D arrangements embedded on the sphere

### PR DESCRIPTION
**Pull request by efifogel**
## Summary of Changes

A stand-alone interactive program that demonstrates 2D arrangements embedded on the sphere. In particular, the program (i) reads a database of all administrative boundaries of the countries in the world, (ii) displays the globe with all countries and land covered by water (which is land not covered by countries) on a window, and (ii) enables interaction with the user. The user can rotate the globe and zoom in and out, and she or he can perform point-location queries to select countries. When a country is selected, it is highlighted. 

This program has potential to grow and support many more features.

This program was developed by Deniz Diktas as part of the Google Summer of Code 2023.

## Release Management

* Affected package(s): Aos_2
* Issue(s) solved (if any):  #7656 , Demonstrating 2D Arrangements Embedded on the Sphere
* Feature/Small Feature (if any):
* Link to compiled documentation (obligatory for small feature) [*wrong link name to be changed*](httpssss://wrong_URL_to_be_changed/Manual/Pkg)
* License and copyright ownership: TAU

## TODO:
- [x]  check branch size (a task for @sloriot)

### CODE REVIEW

**Comment by sloriot** [comment]
Note that there is a conflict with `master`

**Comment by efifogel** [comment]
Right.
I was about to send a message to you indicating that I still need to merge
with master and add a sentence to the CHANGES file.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

File: Arrangement_on_surface_2/demo/earth/CMakeLists.txt [hunk 1]
```diff
@@ -10,90 +12,79 @@ if(POLICY CMP0071)
...
+if (NOT CGAL_FOUND OR NOT CGAL_Qt6_FOUND OR NOT Qt6_FOUND OR NOT Boost_FOUND)
+  if (NOT CGAL_FOUND)
+    set(MISSING_DEPS "the CGAL library, ${MISSING_DEPS}")
+  endif()
+  if (NOT CGAL_Qt6_FOUND)
+    set(MISSING_DEPS "the CGAL Qt6 component, ${MISSING_DEPS}")
+  endif()
+  if (NOT CGAL_Qt6_FOUND)
+    set(MISSING_DEPS "the Qt6 library, ${MISSING_DEPS}")
+  endif()
```
Comment (lrineau) [inline Arrangement_on_surface_2/demo/earth/CMakeLists.txt:28]:
```suggestion
  if (NOT CGAL_Qt6_FOUND)
    set(MISSING_DEPS "the CGAL Qt6 component, ${MISSING_DEPS}")
  endif()
  if (NOT Qt6_FOUND)
    set(MISSING_DEPS "the Qt6 library, ${MISSING_DEPS}")
  endif()
```

Comment (efifogel) [reply Arrangement_on_surface_2/demo/earth/CMakeLists.txt]: thank you; did that of course

**Comment by sloriot** [comment]
- `earth/GUI_event_handler.h:28:47: warning: unused parameter ‘e’ [-Wunused-parameter]`
- `earth/GUI_event_handler.h:29:46: warning: unused parameter ‘e’ [-Wunused-parameter]`
- `earth/GUI_event_handler.h:30:49: warning: unused parameter ‘e’ [-Wunused-parameter]`
- `earth/GUI_event_handler.h:31:27: warning: unused parameter ‘w’ [-Wunused-parameter]`
- `earth/GUI_event_handler.h:31:34: warning: unused parameter ‘h’ [-Wunused-parameter]`
- `earth/Aos.cpp:100:17: warning: ‘std::ostream& {anonymous}::operator<<(std::ostream&, const Approximate_point_2&)’ defined but not used [-Wunused-function]`
- `earth/Aos.cpp:1010:13: warning: unused variable ‘xcv’ [-Wunused-variable]`
- `earth/Aos.cpp:112:17: warning: ‘std::ostream& {anonymous}::operator<<(std::ostream&, const Approximate_Vector_3&)’ defined but not used [-Wunused-function]`
- `earth/Aos.cpp:1248:11: warning: typedef ‘using Exact_type = typename Kernel::FT::Exact_type’ locally defined but not used [-Wunused-local-typedefs]`
- `earth/Aos.cpp:1338:11: warning: typedef ‘using DOuter_ccb = typename Arr_accessor::Dcel_outer_ccb’ locally defined but not used [-Wunused-local-typedefs]`
- `earth/Aos.cpp:1339:11: warning: typedef ‘using DInner_ccb = typename Arr_accessor::Dcel_inner_ccb’ locally defined but not used [-Wunused-local-typedefs]`
- `earth/Aos.cpp:1340:11: warning: typedef ‘using DIso_vert = typename Arr_accessor::Dcel_isolated_vertex’ locally defined but not used [-Wunused-local-typedefs]`
- `earth/Aos.cpp:232:12: warning: variable ‘oi2’ set but not used [-Wunused-but-set-variable]`
- `earth/Aos.cpp:267:10: warning: variable ‘oi2’ set but not used [-Wunused-but-set-variable]`
- `earth/Aos.cpp:355:11: warning: unused variable ‘d’ [-Wunused-variable]`
- `earth/Aos.cpp:360:30: warning: suggest braces around empty body in an ‘if’ statement [-Wempty-body]`
- `earth/Aos.cpp:364:14: warning: unused variable ‘p2’ [-Wunused-variable]`
- `earth/Aos.cpp:481:11: warning: unused variable ‘d’ [-Wunused-variable]`
- `earth/Aos.cpp:486:30: warning: suggest braces around empty body in an ‘if’ statement [-Wempty-body]`
- `earth/Aos.cpp:490:14: warning: unused variable ‘p2’ [-Wunused-variable]`
- `earth/Aos.cpp:541:9: warning: typedef ‘using Face_handle = CGAL::Arrangement_on_surface_2<CGAL::Arr_geodesic_arc_on_sphere_traits_2<CGAL::Epeck>, CGAL::Arr_spherical_topology_traits_2<CGAL::Arr_geodesic_arc_on_sphere_traits_2<CGAL::Epeck>, CGAL::Arr_extended_dcel<CGAL::Arr_geodesic_arc_on_sphere_traits_2<CGAL::Epeck>, {anonymous}::Flag, {anonymous}::Flag, {anonymous}::Flag> > >::Face_handle’ locally defined but not used [-Wunused-local-typedefs]`
- `earth/Aos.cpp:584:27: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<int>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]`
- `earth/Aos.cpp:626:13: warning: unused variable ‘num_new_faces’ [-Wunused-variable]`
- `earth/Aos.cpp:699:9: warning: typedef ‘using Halfedge_handle = CGAL::Arrangement_on_surface_2<CGAL::Arr_geodesic_arc_on_sphere_traits_2<CGAL::Epeck>, CGAL::Arr_spherical_topology_traits_2<CGAL::Arr_geodesic_arc_on_sphere_traits_2<CGAL::Epeck>, CGAL::Arr_extended_dcel<CGAL::Arr_geodesic_arc_on_sphere_traits_2<CGAL::Epeck>, {anonymous}::Flag, {anonymous}::Flag, {anonymous}::Flag> > >::Halfedge_handle’ locally defined but not used [-Wunused-local-typedefs]`
- `earth/Aos.cpp:736:27: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<int>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]`
- `earth/Aos.cpp:778:13: warning: unused variable ‘num_new_faces’ [-Wunused-variable]`
- `earth/Aos.cpp:93:17: warning: ‘std::ostream& {anonymous}::operator<<(std::ostream&, const Dir3&)’ defined but not used [-Wunused-function]`
- `earth/Aos_triangulator.cpp:188:9: warning: typedef ‘using Point = CGAL::Constrained_Delaunay_triangulation_2<CGAL::Epick, CGAL::Triangulation_data_structure_2<CGAL::Triangulation_vertex_base_2<CGAL::Epick>, CGAL::Constrained_triangulation_face_base_2<CGAL::Epick> >, CGAL::Exact_predicates_tag>::Point’ locally defined but not used [-Wunused-local-typedefs]`
- `earth/Aos_triangulator.cpp:45:17: warning: ‘std::ostream& {anonymous}::operator<<(std::ostream&, const Dir3&)’ defined but not used [-Wunused-function]`
- `earth/Aos_triangulator.cpp:51:17: warning: ‘std::ostream& {anonymous}::operator<<(std::ostream&, const Approximate_point_2&)’ defined but not used [-Wunused-function]`
- `earth/Aos_triangulator.cpp:73:9: warning: typedef ‘using Point = CGAL::Constrained_Delaunay_triangulation_2<CGAL::Epick, CGAL::Triangulation_data_structure_2<CGAL::Triangulation_vertex_base_2<CGAL::Epick>, CGAL::Constrained_triangulation_face_base_2<CGAL::Epick> >, CGAL::Exact_predicates_tag>::Point’ locally defined but not used [-Wunused-local-typedefs]`
- `earth/Camera_manip_rot_bpa.cpp:18:59: warning: unused parameter ‘e’ [-Wunused-parameter]`
- `earth/Camera_manip_rot_bpa.cpp:24:58: warning: unused parameter ‘e’ [-Wunused-parameter]`
- `earth/Camera_manip_rot_bpa.cpp:55:61: warning: unused parameter ‘e’ [-Wunused-parameter]`
- `earth/Camera_manip_rot.cpp:14:54: warning: unused parameter ‘e’ [-Wunused-parameter]`
- `earth/Camera_manip_zoom.cpp:17:55: warning: unused parameter ‘e’ [-Wunused-parameter]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘AllButtons’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘BackButton’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton10’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton11’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton12’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton13’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton14’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton15’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton16’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton17’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton18’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton19’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton1’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton20’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton21’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton22’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton23’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton24’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton2’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton3’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton4’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton5’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton6’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton7’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton8’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ExtraButton9’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘ForwardButton’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘MaxMouseButton’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘MouseButtonMask’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘NoButton’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘RightButton’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘TaskButton’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘XButton1’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.cpp:15:10: warning: enumeration value ‘XButton2’ not handled in switch [-Wswitch]`
- `earth/GUI_event_handler.h:28:47: warning: unused parameter ‘e’ [-Wunused-parameter]`
- `earth/GUI_event_handler.h:29:46: warning: unused parameter ‘e’ [-Wunused-parameter]`
- `earth/GUI_event_handler.h:30:49: warning: unused parameter ‘e’ [-Wunused-parameter]`
- `earth/GUI_event_handler.h:31:27: warning: unused parameter ‘w’ [-Wunused-parameter]`
- `earth/GUI_event_handler.h:31:34: warning: unused parameter ‘h’ [-Wunused-parameter]`
- `earth/Kml_reader.cpp:244:27: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<Kml::Node>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]`
- `earth/Kml_reader.cpp:273:29: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<Kml::Node>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]`
- `earth/Line_strips.cpp:105:21: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<unsigned int>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]`
- `earth/Main_widget.cpp:98:44: warning: unused parameter ‘event’ [-Wunused-parameter]`
- `earth/Sphere.cpp:56:21: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<QVector3D>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]`
- `earth/Triangles.cpp:26:7: warning: unused variable ‘num_triangles’ [-Wunused-variable]`
- `earth/Triangles.cpp:30:21: warning: comparison of integer expressions of different signedness: ‘int’ and ‘std::vector<QVector3D>::size_type’ {aka ‘long unsigned int’} [-Wsign-compare]`

**Comment by sloriot** [comment]
First Error:
```
earth/Main_widget.cpp:64:38: error: ambiguous overload for ‘operator<<’ (operand types are ‘QDebug’ and ‘const std::string’ {aka ‘const std::__cxx11::basic_string<char>’})
   64 |     qDebug() << "SELECTED COUNTRY: " << country_name;
```

**Comment by sloriot** [comment]
I would be in favor to rename `earth` into `Arrangement_on_sphere_earth` or something like that

**Comment by sloriot** [comment]
If the .json file is used by only that demo, it can be put in the demo directory

**Comment by sloriot** [comment]
and the license of the file should be mentioned clearly

**Comment by sloriot** [comment]
Almost done:
```
Arrangement_on_surface_2/demo/Arrangement_on_surface_2_earth/GUI_event_handler.h:31:27: warning: unused parameter ‘w’ [-Wunused-parameter]
   31 |   virtual void resize(int w, int h) {}
      |                       ~~~~^
Arrangement_on_surface_2/demo/Arrangement_on_surface_2_earth/GUI_event_handler.h:31:34: warning: unused parameter ‘h’ [-Wunused-parameter]
   31 |   virtual void resize(int w, int h) {}
```

**Comment by efifogel** [comment]
ok, I fixed everything I could.
1. Regarding the error, I tried an alternative syntax. I hope it works.
2. Regarding the renaming, I renamed the directory, but retained the name
of the executable (earth). Do you think it's ok?
3. Regarding the license of the json file, I added the record below (to the
json file).
I'm not sure if anything else is needed.
"description": "Based on Natural Earth data set (Free vector and raster map
data @ naturalearthdata.com). 1:110m scale."
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by sloriot** [comment]
> If the .json file is used by only that demo, it can be put in the demo directory

Actually it could be used by triangulation on sphere, so a better name than arrangement should be picked for the data directory.

**Comment by efifogel** [comment]
oh, yes, I may use it for other purposes as well. Perhaps just remove the
"arrangement _3"  sub dir?

**Comment by efifogel** [comment]
Any idea why?
I commented out the (probably) erroneous "install" statement in the
CMakeLists.txt file, but I doubt that this was not the reason for the
failure.
While at it, I fixed one warning that showed up.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by sloriot** [comment]
maybe the qt specific instructions?

**Comment by efifogel** [comment]
I added the 3 statements below, which I found in the standard demo.
It's a guess...

add_to_cached_list(CGAL_EXECUTABLE_TARGETS earth)
include(${CGAL_MODULES_DIR}/CGAL_add_test.cmake)
cgal_add_compilation_test(earth)
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by lrineau** [comment]
> Any idea why?

The error is:
```
-- Configuring test in test/Arrangement_on_surface_2_earth_Demo
CMake Error at test/Arrangement_on_surface_2_earth_Demo/CMakeLists.txt:15 (find_package):
  find_package for module Qt6 called with REQUIRED, but
  CMAKE_DISABLE_FIND_PACKAGE_Qt6 is enabled.  A REQUIRED package cannot be
  disabled.
```

`find_package(Qt6)` is used with the `REQUIRED` option. So, that triggered an error on all test platforms without Qt6 (and they are numerous).

File: Arrangement_on_surface_2/demo/Arrangement_on_surface_2_earth/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+#SOURCE FILES (NOT CATEGORIZED YET)
+file(GLOB source_files Common_defs.h earth.cpp Timer.h
+  Main_widget.h Main_widget.cpp
+  Message_manager.h Message_manager.cpp
+  Tools.h Tools.cpp
+  Verification.h Verification.cpp)
+source_group("Source Files" FILES ${source_files})
+
+qt_standard_project_setup()
```
Comment (sloriot) [inline Arrangement_on_surface_2/demo/Arrangement_on_surface_2_earth/CMakeLists.txt:90]: this was introduced in QT 6.3

**Comment by efifogel** [comment]
Thanks.
Removed the "REQUIRED" option in find_package(Qt6...)
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

**Comment by sloriot** [comment]
Still failing in CGAL-6.0-Ic-191

**Comment by efifogel** [comment]
The demo had some warnings, which I fixed.
   ____  _        ____             _
  /_____/_) o    /__________  __  //
 (____ (   (    (    (_/ (_/-(-'_(/
                         _/

File: Arrangement_on_surface_2/demo/Arrangement_on_surface_2_earth/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+  if (NOT Boost_FOUND)
+    set(MISSING_DEPS "the Boost library, ${MISSING_DEPS}")
+  endif()
+
+  message(STATUS "This project requires ${MISSING_DEPS} and will not be compiled.")
+  return()
+endif()
+
+################################################################################
+include(FetchContent)
```
Comment (sloriot) [inline Arrangement_on_surface_2/demo/Arrangement_on_surface_2_earth/CMakeLists.txt:37]: @lrineau this is the first cmake script in CGAL where we use fetch content. Are we OK with using it?

Comment (sloriot) [reply Arrangement_on_surface_2/demo/Arrangement_on_surface_2_earth/CMakeLists.txt:37]: It actually seems to be responsible for the white columns in 6.0-Ic-194

**Comment by sloriot** [comment]
There are also some files put in the source directory:
```
.qt/
_deps/
earth
earth_autogen/
```

File: Arrangement_on_surface_2/test/Arrangement_on_surface_2/Traits_test.h [hunk 4]
```diff
@@ -1115,25 +1122,70 @@ Traits_test<Geom_traits_T>::merge_wrapper_imp(std::istringstream& str_stream,
   return this->compare_curves(this->m_xcurves[id], cv);
 }
 
+//! Fall threw instance
+template <typename GeomTraits>
+template <typename GT>
+bool Traits_test<GeomTraits>::
+approximate_wrapper_impl2(std::istringstream& is, long) { return false; }
```
Comment (sloriot) [inline Arrangement_on_surface_2/test/Arrangement_on_surface_2/Traits_test.h:1129]:
```suggestion
approximate_wrapper_impl2(std::istringstream&, long) { return false; }
```

File: Arrangement_on_surface_2/test/Arrangement_on_surface_2/Traits_test.h [hunk 5]
```diff
@@ -1115,25 +1122,70 @@ Traits_test<Geom_traits_T>::merge_wrapper_imp(std::istringstream& str_stream,
...
+    return false;
+  }
+  return true;
+}
+
+//! Fall threw instance
+template <typename GeomTraits>
+template <typename GT>
+bool Traits_test<GeomTraits>::
+approximate_wrapper_impl1(std::istringstream& is, long) { return false; }
```
Comment (sloriot) [inline Arrangement_on_surface_2/test/Arrangement_on_surface_2/Traits_test.h:1167]:
```suggestion
approximate_wrapper_impl1(std::istringstream&, long) { return false; }
```

**Comment by sloriot** [comment]
There are warnings in CGAL-6.0-Ic-205. You can visit [that page](https://cgal.geometryfactory.com/CGAL/testsuite/summary-6.0-Ic-205.html) and you'll get a summary of errors and warnings.

File: Arrangement_on_surface_2/examples/Arrangement_on_surface_2/spherical_insert.cpp [hunk 6]
```diff
@@ -13,39 +15,59 @@
...
+  arcs.push_back(ctr_xcv(ctr_p(0, -1, 0), ctr_p(0, 0, -1)));
+  arcs.push_back(ctr_xcv(ctr_p(0, -1, 0), ctr_p(0, 0, 1)));
+  arcs.push_back(ctr_xcv(ctr_p(1, 0, 0), ctr_p(0, 1, 0)));
+  arcs.push_back(ctr_xcv(ctr_p(1, 0, 0), ctr_p(0, -1, 0)));
+  arcs.push_back(ctr_xcv(ctr_p(-1, 0, 0), ctr_p(0, 1, 0)));
+  arcs.push_back(ctr_xcv(ctr_p(-1, 0, 0), ctr_p(0, -1, 0)));
+
+  // Construct meridians
+  std::size_t num_meridians(16);
+  if (argc > 1) sscanf(argv[1], "%zu", &num_meridians);
```
Comment (sloriot) [inline Arrangement_on_surface_2/examples/Arrangement_on_surface_2/spherical_insert.cpp:51]:
```
         C:\CGAL_ROOT\CGAL-6.0-Ic-206\test\Arrangement_on_surface_2_Examples\spherical_insert.cpp(51,17): warning C4996: 'sscanf': This function or variable may be unsafe. Consider using sscanf_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details. [C:\CGAL_ROOT\CGAL-6.0-Ic-206\cmake\platforms\MSVC-2022-Community-Release\test\Arrangement_on_surface_2_Examples\spherical_insert.vcxproj]
```

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.0-Ic-21[56].

However the size of the branch is 11MB. I need to have a closer look

**Comment by sloriot** [comment]
@lrineau  `ne_110m_admin_0_countries.json` is 7.7M, (929K compressed) and the branch is 1.2M.

**Comment by sloriot** [comment]
```
/mnt/testsuite/test/Arrangement_on_surface_2_earth_Demo/Aos.cpp:1333:9: warning: typedef ‘using Point_2 = CGAL::Arrangement_on_surface_2, CGAL::Arr_spherical_topology_traits_2, CGAL::Arr_face_extended_dcel, std::__cxx11::basic_string > > >::Point_2’ locally defined but not used [-Wunused-local-typedefs]
/mnt/testsuite/test/Arrangement_on_surface_2_earth_Demo/Aos_triangulator.cpp:224:11: warning: typedef ‘using Faces_ = class std::vector >’ locally defined but not used [-Wunused-local-typedefs]
```

**Comment by efifogel** [comment]
Regarding my previous commit, I had a moment and tried the demo on a much more simple faked map, and noticed that the rendering was busted, so I fixed it.
I don't plan to make more changes in this PR.
Thanks for your last fix of the warnings.

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.0-Ic-228

(@lrineau  you did not react to https://github.com/CGAL/cgal/pull/7678#issuecomment-2057320464)

**Comment by lrineau** [comment]
> @lrineau `ne_110m_admin_0_countries.json` is 7.7M, (929K compressed) and the branch is 1.2M.

Well, as far as I understand, there is nothing we can do. We will keep it as it is.

**Pull request merged**